### DISCUSSION TITLE
Changing the HTML WYSIWYG editor - What is involved?

### DISCUSSION THREAD

**Discussion by jgruttad92** [OrchardCMS/OrchardCore#14771] [Q&A]
Hi all,

I would like to override my instance of Orchard Core CMS such that it uses TinyMCE instead of Trumbowyg when writing HTML. I see on the [HTML Orchard Core Page](https://docs.orchardcore.net/en/main/docs/reference/modules/Html/#custom-editors) that it's possible to add/use different editors, but I have yet to find a good example of doing so online.

So ultimately I was wondering if anyone had done something like this before, and would be willing to impart their wisdom and show what they needed to do (for TinyMCE or some other HTML WYSIWYG editor).

Thanks,
- Jeff G

**Comment by ns8482e** [comment]
You can change the editor by defining your own editor option and implementing it. 

Like to define editor option create HtmlBodyPart-MyEditor.Option.cshtml

And then implement your editor using HtmlBodyPart-MyEditor.Edit.cshtml

**Reply by jgruttad92** [reply]
I've followed this so far, using the sample custom editor from the docs. Is there an additional step required to allow it to be used? Worth noting: My editor flagged 'Body' as an invalid property from the following snippet, so I've tried using Html and HtmlBodyPart instead, with no such luck:

```
@using OrchardCore.Html.ViewModels
@model HtmlBodyPartViewModel

<fieldset class="mb-3">
    <label asp-for="Body">@T["Body"]</label>
    <textarea asp-for="Body" rows="5" class="form-control"></textarea>
    <span class="hint">@T["The body of the content item."]</span>
</fieldset>
```

The files I created were HtmlBodyPart-Tiny.Option.cshtml and HtmlBodyPart-Tiny.Edit.cshtml

The option file itself looks like:
```
@{
    string currentEditor = Model.Editor;
}
<option value="Tiny" selected="@(currentEditor == "Tiny")">@T["Tiny Editor"]</option>
```

**Reply by ns8482e** [reply]
I guess with defining this two files you can change the editor for html body part on your content type to Tiny

**Reply by jgruttad92** [reply]
Unfortunately, I still don't see the option:
![image](https://github.com/OrchardCMS/OrchardCore/assets/42447606/96c6f2c9-3c83-4e9e-ba0e-26b7cd1f0a1b)
![image](https://github.com/OrchardCMS/OrchardCore/assets/42447606/b505787b-a33b-4633-96e8-30c97687688b)

**Reply by ns8482e** [reply]
RawHtml uses HtmlField not HtmlBodyPart. Change your file names to HtmlField-Tiny.Options.cshtml and HtmlField-Tiny.Edit.cshtml

**Reply by jgruttad92** [reply]
Understood. This is how I have it set up now:
![image](https://github.com/OrchardCMS/OrchardCore/assets/42447606/bd2f7bf4-4fe8-4e1c-b4b5-a7ba543468fe)

There wasn't an example/sample of an HTML Field editor, so I kept the same structure from the HtmlBodyPart-Tiny.Edit.cshtml file.

Unfortunately, I still cannot see it when I attempt to to modify the editor for RawHtml:
![image](https://github.com/OrchardCMS/OrchardCore/assets/42447606/4e803a7a-4fd5-4326-8cb5-d0aefc2230a8)

**Reply by ns8482e** [reply]
You added views in a module, so make sure views on your modules override the default

Check following 

- module target is referenced in your module project
-  Views are set as embedded resources and not content 
- your module has dependency of ContentField defined so the views can override the views of ContentField
- and your module is enabled

**Reply by jgruttad92** [reply]
I ended up moving the views to the web server project, and changed them from content to embedded resource. I now get an error when I go to the page to edit which HTML field editor is used:
![image](https://github.com/OrchardCMS/OrchardCore/assets/42447606/c78bda43-7dde-4600-b0e7-684467be9773)

**Reply by ns8482e** [reply]
I guess In main web project set it as content, 

Views in module are harvested from Embedded resources
